department-of-veterans-affairs/caseflow
# Allow Selection of Open Tasks during Pending-Appeal Substitution

Connects [CASEFLOW-1786](https://vajira.max.gov/browse/CASEFLOW-1786)

### Description
This updates the appellant substitution UI to show open (`assigned` & `on_hold`) tasks; the user can select one or more tasks to cancel as part of the substitution.

### Acceptance Criteria
- [ ] Copy and layout matches design
- [ ] Deselecting an open parent task deselects and disables associated child task(s)

### Testing Plan
1. Log in as `COB_USER`
2. Search cases for vet file number `54545454` and choose one of the pending appeals (those w/o `Dispatched` status)
3. Begin substitution and proceed to tasks form
4. Verify that active tasks section can be viewed and matches design
5. Verify that deselecting a parent task (like `JudgeDecisionReviewTask`) will deselect and disable child tasks (Such as `AttorneyTask`)

### User Facing Changes
![Screen Shot 2021-11-01 at 12 58 22](https://user-images.githubusercontent.com/2581274/139710183-e12c3427-734e-4bca-9922-fe7c08840827.png)
